src/pytorch_lightning/utilities/parameter_tying.py

```diff
@@ -44,8 +41,7 @@ def _find_shared_parameters(module: nn.Module, tied_parameters: Optional[Dict] =
             continue
         submodule_prefix = prefix + ("." if prefix else "") + name
         _find_shared_parameters(m, tied_parameters, submodule_prefix)
-    if first_call:
```

Comment:
I don't think this was required, other than for the improvement in efficiency of not returning anything for the inner recursive calls